webhook/resources/backup/validator.go

```diff
@@ -67,5 +64,21 @@ func (b *backupValidator) Create(request *admission.Request, newObj runtime.Obje
 		return werror.NewInvalidError(fmt.Sprintf("backup target %s is not available", backupTargetName), "")
 	}
 
+	//check if label volume name matches snapshot volume name
+	labelVolumeName := backup.Labels[types.LonghornLabelBackupVolume]
+	snapshot, _ := b.ds.GetSnapshot(backup.Spec.SnapshotName)
```

Comment:
Check the returned `err`; if the snapshot is missing, the next line could lead to invalid memory access.

```suggestion
	snapshot, _ := b.ds.GetSnapshot(backup.Spec.SnapshotName)
```